requirements.txt

```diff
@@ -1,34 +1,23 @@
-bump2version==0.5.11
-wheel==0.33.6
-watchdog==0.9.0
-flake8==3.8.4
-mypy==0.790
-tox==3.14.0
-coverage==4.5.4
-Sphinx==1.8.5
-twine==1.14.0
-Click==7.1
+bump2version
+wheel
+watchdog
+flake8
+mypy
+tox
+coverage
+twine
 f90nml>=1.1.0
 appdirs>=1.4.0
-requests
 sphinx_rtd_theme
-pytest==5.2.2
-pytest-cov==2.8.1
-pytest-subtests==0.3.0
+pytest-cov
+pytest-subtests
 gcsfs>=0.7.0
 google-cloud-storage
-fsspec==0.8.0
-kubernetes==10.0.1
-black==19.10b0
-xarray==0.15.1
-numpy==1.16.5
-numcodecs==0.7.2
-zarr==2.3.2
-h5py==2.10.0
-h5netcdf==0.8.0
-dask==2.10.1
-toolz==0.10.0
-scipy==1.3.1
+black
+numcodecs==0.7.2 #pin for gt4py, h5py and py3.6 to agree
+h5py==2.10.0 #pin for gt4py, h5py and py3.6 to agree
+h5netcdf
+dask==2021.03.0 #py3.6 - last supported version
+numpy==1.17.* #pin for gt4py, h5py, cupy9.1 and py3.6 to agree
```

Comment:
Can you make the "pinned" versions a looser requirement (e.g. a >= and/or <=?)? You should not need to "resolve dependencies" yourself, though you do still need something here.

Side rant, Ideally pip-compile can read gt4py's dependencies and include those when resolving the pinned versions. However it seems it can't, either because it's a github link or because the requirements are in a [setup.cfg file](https://github.com/GridTools/gt4py/blob/master/setup.cfg). This is made even harder by the fact that gt4py's requirements are much more strict than gt4py actually requires - for example, it requires numpy 1.15.* in the setup.cfg, even though we're using it with much newer numpy versions. This is probably because we're pinning gt4py dependencies manually and in the setup.py/cfg file, instead of automatically and in a place different from the package requirements (e.g. a constraints.txt).

I'd encourage that we use saner requirements in gt4py to solve some of this, e.g. so we could copy-paste the requirements from its setup.cfg to our requirements.txt and it would make sense.

Reply:
It looks like I misread some of the gt4py requirements (I misinterpreted `~=`), for numpy it says >=1.15, <2. You may be able to copy those requirements here.

Reply:
Copying `gt4py` reqs for `numpy` did the trick